### PR TITLE
WEBAPP-547: Use test cms for dev builds

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

File: src/modules/app/fetchData.js [hunk 1]
```diff
@@ -28,19 +26,7 @@ async function fetchData<P, T> (
...
-    }
-    if (responseOverride) {
-      const data = endpoint.mapResponse(responseOverride, params)
-      const payload = new Payload(false, formattedUrl, data, null)
-      dispatch(finishFetchAction(endpoint.stateName, payload))
-      return payload
-    }
-
-    const payload = await endpoint.request(params, formattedUrl)
+    const payload = await endpoint.request(params)
```
Comment (maxammann) [inline src/modules/app/fetchData.js:29]: Whouls we create a task to remove the errorOverride functionality?

Comment (maxammann) [reply src/modules/app/fetchData.js:29]: I think it is  not used in the RN app. Not sure though.

Comment (klinzo) [reply src/modules/app/fetchData.js:29]: This is all a duplicate of code present in the api-client. Therefore I thought I'd just remove it.

Comment (klinzo) [reply src/modules/app/fetchData.js:29]: It is used for some tests, so I don't think we should remove the functionality entirely.

File: package.json [hunk 2]
```diff
@@ -7,9 +7,9 @@
     "npm": ">=3.8"
   },
   "scripts": {
-    "start": "webpack-dev-server --config tools/webpack.config.babel.js --progress --env.prod=false --env.config_name=integreat",
+    "start": "webpack-dev-server --config tools/webpack.config.babel.js --progress --env.prod=false --env.config_name=integreat-debug",
```
Comment (Taggotty) [inline package.json:10]: So the regular start command disappears. Is it not used anywhere?

Comment (klinzo) [reply package.json:10]: No, this is just a debug command starting the local webpack server. Also, if you want to use the live cms with the start command, you can still set the url manually back to `cms.integreat-app.de`

File: tools/integreat-debug-config/index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import brightTheme from '../../src/modules/theme/constants/theme'
+
+const IntegreatAppDebugConfig: AppConfigType = {
+  appTitle: 'Integreat',
+  itunesAppId: '1072353915',
+  theme: brightTheme,
+  cmsUrl: 'https://cms-test.integreat-app.de',
+  locationIcon: '/location-big.svg',
+  logoWide: '/integreat-app-logo.png',
+  internalLinksHijackPattern: 'https?:\\/\\/(cms\\.integreat-app\\.de|web\\.integreat-app\\.de|integreat\\.app)(?!\\/[^/]*\\/(wp-content|wp-admin|wp-json)\\/.*).*'
```
Comment (Schedulaar) [inline tools/integreat-debug-config/index.js:14]: I would suggest to adjust this to
```
internalLinksHijackPattern: 'https?:\\/\\/(cms-test\\.integreat-app\\.de|web\\.integreat-app\\.de|integreat\\.app)(?!\\/[^/]*\\/(wp-content|wp-admin|wp-json)\\/.*).*'
```

**Pull request merged**